Cleaned TS yaml rules

reaperrr opened the pull request:
Streamlines actor defaults.

GraionDilach commented:
You coulda enable the EMP sparkle anim as part of this iMO - now that we have WithDecoration with HCenter-VCenter positioning.

abcdefg30 commented on mods/ts/rules/defaults.yaml:
```diff
@@ -271,50 +258,57 @@
+			Prone100Percent: 100
+			Prone350Percent: 350
+		DamageTriggers: TriggerProne
+	TakeCover:
```
Another `TakeCover:`? (see line 272)

reaperrr replied:
Silly copy-paste error that I somehow overlooked.

abcdefg30 commented:
Needs a rebase.

reaperrr commented:
Rebased.

reaperrr committed:
Cleaned TS yaml rules.

Streamlined actor defaults, removed a lot of duplication.

reaperrr committed:
Fixed veinhole sequence offset

reaperrr commented:
I want to split off the third commit into a separate PR that will deal with civ building sequence offsets as well.

pchote commented:
The mammoth mk II and carryall selection boxes are weird, but everything else looks fine.  :+1: after fixing those.

reaperrr committed:
Fixed Carryall and Mammoth Mk.II selection boxes.

Removed now-redundant custom selection boxes from Wolverine and Titan.

reaperrr commented:
Updated.

![fixed-selectionboxes](https://cloud.githubusercontent.com/assets/2857877/8293458/d080103c-1935-11e5-9372-034472fd27d5.png)

obrakmann commented:
I suspect this means something's wrong with the viceroids:

```
Exception of type `System.InvalidOperationException`: Actor has multiple traits of type `OpenRA.Mods.Common.Traits.RenderSprites`
  at OpenRA.TraitDictionary+TraitContainer`1[OpenRA.Mods.Common.Traits.RenderSprites].GetOrDefault (UInt32 actor) [0x000b4] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:161 
  at OpenRA.TraitDictionary+TraitContainer`1[OpenRA.Mods.Common.Traits.RenderSprites].Get (UInt32 actor) [0x00003] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:146 
  at OpenRA.TraitDictionary.Get[RenderSprites] (OpenRA.Actor actor) [0x00013] in /home/oliver/devel/openra/git/OpenRA.Game/TraitDictionary.cs:96 
  at OpenRA.Actor.Trait[RenderSprites] () [0x0000d] in /home/oliver/devel/openra/git/OpenRA.Game/Actor.cs:212 
  at OpenRA.Mods.Common.Traits.WithSpriteBody..ctor (OpenRA.ActorInitializer init, OpenRA.Mods.Common.Traits.WithSpriteBodyInfo info, System.Func`1 baseFacing) [0x00027] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:55 
  at OpenRA.Mods.Common.Traits.WithSpriteBody..ctor (OpenRA.ActorInitializer init, OpenRA.Mods.Common.Traits.WithSpriteBodyInfo info) [0x00000] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:50 
  at OpenRA.Mods.Common.Traits.WithSpriteBodyInfo.Create (OpenRA.ActorInitializer init) [0x00003] in /home/oliver/devel/openra/git/OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:29 
  at OpenRA.Actor..ctor (OpenRA.World world, System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x000c8] in /home/oliver/devel/openra/git/OpenRA.Game/Actor.cs:95 
  at OpenRA.World.CreateActor (Boolean addToWorld, System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x00004] in /home/oliver/devel/openra/git/OpenRA.Game/World.cs:241 
  at OpenRA.World.CreateActor (System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x00005] in /home/oliver/devel/openra/git/OpenRA.Game/World.cs:236 
  at OpenRA.Mods.Cnc.Traits.SpawnViceroid+<Killed>c__AnonStorey0.<>m__0 (OpenRA.World w) [0x0006e] in /home/oliver/devel/openra/git/OpenRA.Mods.Cnc/Traits/SpawnViceroid.cs:51
```

penev92 commented on mods/ts/rules/defaults.yaml:
```diff
@@ -376,84 +373,40 @@
+
+^CivilianVoxelVehicle:
+	Inherits: ^VoxelVehicle
+	-RenderSprites: # They don't have icons so this is needed to satisfy --check-yaml
```
I vote for avoiding comments in YAML wherever possible, because the utility doesn't like them and keeps removing them. Also I don't think this is exactly needed here.

penev92 commented on mods/ts/rules/defaults.yaml:
```diff
@@ -1,3 +1,14 @@
+	AppearsOnRadar:
+	UpdatesPlayerStatistics:
+	CombatDebugOverlay:
+	DrawLineToTarget:
```
This looks like an overkill, considering not even half of the actors that inherit this will need target lines.

reaperrr committed:
Several fixes

Fixed TS building repairability to match original.
Civilian structures can now only be repaired with engineers.

Removed ProximityCaptor.

Various other fixes.

reaperrr commented:
Updated.

- fixed visceroid crash
- removed comment from civ voxel vehicle default
- moved DrawLineToTarget to moving actors so buildings won't inherit it

- slightly reorganised aircraft defaults

reaperrr commented:
I'll squash the last commit when this has its 2nd +1.

GraionDilach commented on mods/ts/rules/defaults.yaml:
```diff
@@ -41,18 +51,33 @@
 		DestroyedSounds: crmble2.aud
 	RenderBuilding:
 	WithBuildingExplosion:
+	RepairableBuilding:
```
This lets civilian buildings be repaired via the mouse. I thought they were only repairable via engineers in TS as well.

reaperrr replied:
You're right, I overlooked that. Will fix it.

GraionDilach commented on mods/ts/rules/defaults.yaml:
```diff
@@ -41,19 +51,34 @@
 	WithBuildingExplosion:
+	EngineerRepairable:
+	ShakeOnDeath:
+	ProximityCaptor:
```
Why is this needed?

penev92 replied:
That one isn't needed in any of the official mods. Some Capture the Flag maps use it, but they can define it for themselves.

reaperrr commented:
Removed ProximityCaptor and squashed the last 3 smaller commits.

Mailaender commented:
> Removes bogus TEMPERATE exclusivity from CA00** civ buildings.

I still don't see them on SNOW.

reaperrr commented:
> I still don't see them on SNOW.

Right, I decided to concentrate on the actor defaults for this PR and moved that fix to an upcoming follow-up PR quite a while ago (that follow-up will fix civilian structure sequence offsets as well), forgot to update the PR description.

Mailaender commented:
Can't spot any regressions in-game. :white_check_mark:

GraionDilach commented:
Even if I come late to the party but I really welcome this cleanup. It looks so much better now. Great job. :)
